python/oneflow/test/xrt/test_reshape.py

```diff
@@ -0,0 +1,43 @@
+"""
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+from random import randint
+from random import choice
+
+import numpy as np
+
+import oneflow as flow
+from test_xrt import *
+
+
+class TestXrtReLU(flow.unittest.TestCase):
+    def test_xrt_relu(test_case):
+        x = np.random.random((1, 10, 2)).astype(np.float32)
+        x_cpu = flow.tensor(x, dtype=flow.float32, device=flow.device("cpu"))
+        x_cuda = flow.tensor(x, dtype=flow.float32, device=flow.device("cuda"))
+
+        reshape = lambda arg: flow.reshape(arg, shape=[2, 2, -1])
+        reshape_g = generate_graph(reshape)
```

Comment:
```suggestion
        generate_reshape_graph = lambda: generate_graph(reshape)
        reshape_g = generate_reshape_graph()
```
这样 37-39 行就可以用 generate_reshape_graph() 代替 generate_graph(reshape) 了。其它几个测试文件也是同理。

Reply:
这里没必要改吧。

Reply:
如果日后对 reshape graph 的需求变化了，generate_graph 满足不了要求，要用其它方式生成 reshape graph，generate_reshape_graph 的调用者就不用再修改代码了

Reply:
我并不觉得lambda表达式用太多是一个好东西。这里我觉得没有必要改

Reply:
有需要的时候，再改代码就行了

Reply:
> 我并不觉得lambda表达式用太多是一个好东西

这是什么原因呢。如果觉得可读性差，写成在 test_xrt_reshape 内部定义的普通函数也可以
```python
def test_xrt_reshape(test_case):
    ....
    def generate_reshape_graph():
        return generate_graph(reshape)
    ....
```
用 generate_reshape_graph 代替 generate_graph(reshape) 之后，我们确实会得到封装性更好、更面向接口的代码

Reply:
就几行代码，一眼就能看懂。何必封装呢。 下一个人，觉得代码冗余了，那就再封装不就行了。

Reply:
唔，目前的写法还是会让读者有一些疑惑的，我看到目前的写法的第一反应是 37-39 行为什么不复用公共子表达式，想一想才明白现在这样写是因为每个测试都要求传入一个全新的 graph。我能明白这一点的原因是我对相关背景（所有的 config 要在 graph 初次运行前设置好，初次运行后就不能修改）有一些了解，所以经过短暂的疑惑之后就能联想的到。如果封装一个无参函数 generate_reshape_graph()，那 ”每次都生成一个全新的 graph“ 的意图就很明确，不需要让读者去悟

Reply:
调用 generate_graph(flow.relu) 和 generate_reshape_graph()都是传递相同的信息，这里生成了一个全新的graph

Reply:
我觉得读者看到 `generate_graph(reshape)` 重复了三遍，第一反应会是 ”代码质量不好，没有提取公共表达式“。专门写一个 generate_reshape_graph()，读者就没有 “作者能提取公共表达式但却没有提取” 的杂念，更容易明白这里在做的事情是在生成一个全新的 graph。

Reply:
这里的本质是代码风格的问题。什么样的代码风格是好的。 和是否生成了一个全新的graph没有关系。  

我觉得  generate_graph(reshape) 这个代码已经是简洁易懂的，没有必要再wrap一层函数。多wrap一层函数，反而增加了理解的成本。

Reply:
那加个注释吧，注明一下这里 `generate_graph(reshape)` 重复三遍是故意的，目的是生成全新的 graph，不然可能哪次重构就误改了